docs/java-rest/high-level/indices/refresh.asciidoc

```diff
@@ -0,0 +1,75 @@
+[[java-rest-high-refresh]]
+=== Refresh API
+
+[[java-rest-high-refresh-request]]
+==== Refresh Request
+
+A `RefreshRequest` can be applied to one or more indices, or even on `_all` the indices:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/IndicesClientDocumentationIT.java[refresh-request]
+--------------------------------------------------
+<1> Refresh one index
+<2> Refresh multiple indices
+<3> Refresh all the indices
+
+[[java-rest-high-refresh-sync]]
+==== Synchronous Execution
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/IndicesClientDocumentationIT.java[refresh-execute]
+--------------------------------------------------
+
+[[java-rest-high-refresh-async]]
+==== Asynchronous Execution
+
+The asynchronous execution of a refresh request requires both the `RefreshRequest`
+instance and an `ActionListener` instance to be passed to the asynchronous
+method:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/IndicesClientDocumentationIT.java[refresh-execute-async]
+--------------------------------------------------
+<1> The `RefreshRequest` to execute and the `ActionListener` to use when
+the execution completes
+
+The asynchronous method does not block and returns immediately. Once it is
+completed the `ActionListener` is called back using the `onResponse` method
+if the execution successfully completed or using the `onFailure` method if
+it failed.
+
+A typical listener for `RefreshResponse` looks like:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/IndicesClientDocumentationIT.java[refresh-execute-listener]
+--------------------------------------------------
+<1> Called when the execution is successfully completed. The response is
+provided as an argument
+<2> Called in case of failure. The raised exception is provided as an argument
+
+[[java-rest-high-refresh-response]]
+==== Refresh Response
+
+The returned `RefreshResponse` allows to retrieve information about the
+executed operation as follows:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/IndicesClientDocumentationIT.java[refresh-response]
+--------------------------------------------------
+<1> total shards number of requested indices
+<2> successfully refreshed shards number
+<3> failed to refresh shards number
+<4> an array of `DefaultShardOperationFailedException` if exist, otherwise it will be an empty array
+
+If the indices were not found, an `ElasticsearchException` will be thrown:
```

Comment:
add "by default" ? that's because you can affect this behaviour through the different indices options.